assignment/todo/cmd/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"todo"
+)
+
+const (
+	fileName = ".todo.json"
+	NoAction = -1
+)
+
+func main() {
+	add := flag.Bool("add", false, "Add a new task")
+	complete := flag.Int("complete", -1, "Complete a task")
+	delete := flag.Int("delete", -1, "Delete a task")
+
+	flag.Parse()
+
+	var list todo.List
+
+	if err := list.Get(fileName); err != nil {
+		fmt.Fprintf(os.Stderr, "Error getting tasks: %v\n", err)
+		os.Exit(1)
+	}
+
+	args := flag.Args()
+
+	if *add || (len(args) > 0 && *complete == NoAction && *delete == NoAction) {
```

Comment:
Puede simplificar el procesamiento si usa `ifs`.  En lugar de incluir múltiples combinaciones de argumentos dentro de la validación, puede ordenar if sencillos en el orden que quiere y dentro del if ejecutar el return. Ejemplo:

```go
// primero el add:
if *add {
  // código
  return
}

if *complete {
  // código
  return
}

// .. más validaciones y código
```

Este es un patrón que se conoce como `early return`. Por aquí dejo un artículo de referencia: https://dev.to/jpswade/return-early-12o5